src/main/java/seedu/address/logic/commands/MarkCommand.java

```diff
@@ -0,0 +1,287 @@
+        AttendanceList updatedAttendanceList = getUpdatedAttendanceList(personToMark, markPersonDescriptor);
+        SessionList sessionList = personToMark.getSessionList();
+        GradeProgressList gradeProgressList = personToMark.getGradeProgressList();
+        Set<Tag> tags = personToMark.getTags();
+
+        return new Person(name, phone, lessonPlan, updatedHomeworkList,
+                updatedAttendanceList, sessionList, gradeProgressList, tags);
+    }
+
+    public static HomeworkList getUpdatedHomeworkList(Person personToEdit, MarkPersonDescriptor markPersonDescriptor)
+            throws CommandException {
+        HomeworkList updatedHomeworkList = personToEdit.getHomeworkList();
+        Optional<Index> homeworkIndex = markPersonDescriptor.getHomeworkIndex();
+        if (homeworkIndex.isEmpty()) {
+            return updatedHomeworkList;
+        }
+        if (!updatedHomeworkList.isValidIndex(homeworkIndex.get())) {
+            throw new CommandException(HomeworkList.MESSAGE_INVALID_HOMEWORK_INDEX);
+        }
+        updatedHomeworkList.markAtIndex(homeworkIndex.get());
+        return updatedHomeworkList;
+    }
+
+    public static AttendanceList getUpdatedAttendanceList(Person personToMark,
+                                                           MarkPersonDescriptor mPerDesc) throws CommandException {
+        AttendanceList updatedAttendanceList = personToMark.getAttendanceList();
+        Optional<Index> attendanceIndex = mPerDesc.getAttendanceIndex();
+        if (attendanceIndex.isEmpty()) {
+            return updatedAttendanceList;
+        }
+        if (!updatedAttendanceList.isValidIndex(attendanceIndex.get())) {
+            throw new CommandException(AttendanceList.MESSAGE_INVALID_ATTENDANCE_INDEX);
+        }
+        updatedAttendanceList.markAtIndex(attendanceIndex.get());
+        return updatedAttendanceList;
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        // short circuit if same object
+        if (other == this) {
+            return true;
+        }
+
+        // instanceof handles nulls
+        if (!(other instanceof MarkCommand)) {
+            return false;
+        }
+
+        // state check
+        MarkCommand e = (MarkCommand) other;
+        return markPersonDescriptor.equals(e.markPersonDescriptor);
+    }
+
+    /**
+     * Stores the details to mark the person with. Each non-empty field value will replace the
+     * corresponding field value of the person.
+     */
+    public static class MarkPersonDescriptor {
+        private Name name;
+        private Phone phone;
+        private LessonPlan lessonPlan;
+        private Index homeworkIndex;
+        private Homework homework;
+        private Index gradeProgressIndex;
+        private GradeProgress gradeProgress;
+        private Index attendanceIndex;
+        private Attendance attendance;
+        private Index sessionIndex;
+        private Session session;
+        private Set<Tag> tags;
```

Comment:
Actually all of these except the Homework and Attendance fields don't seem necessary since the rest aren't edited by MarkCommand. This is probably the same for RemoveCommand too

It still seems functional (just wasteful) so you can fix this in v1.4 if you'd rather merge this change first.

Reply:
yeap will address this during v1.4